doc/fr/developper/writestreamconnector.rst

```diff
@@ -0,0 +1,586 @@
+
+`InfluxDB <https://www.influxdata.com/>`_ is a Time Series database. We will use 
+this storage to insert performance data collected by the Centreon platform. For 
+this example, we will use the predefined `InfluxDB Docker <https://hub.docker.com/_/influxdb/>`_.
+
+To send data to InfluxDB, we need parameters to access to InfluxDB storage:
+
+* **http_server_address**: IP address of the storage
+* **http_server_port**: 8086 by default
+* **http_server_protocol**: http or https
+* **influx_database**: name of database
+* **influx_user**: user to access to database if defined
+* **influx_password**: password of user to access to database if defined
+
+In order to do not saturate the storage, we will add all events in a queue and 
+send data by bulk. We need to define the size of the queue and the maximum 
+delay before send events:
+
+* max_buffer_size  
+* max_buffer_age
+
+The LUA script will contain:
+
+A function to create the queue
+******************************
+
+This function initialises the queue with default parameters or defined by the 
+user. Moreover, the queue settings and the events will be store into a 
+dedicated structure:
+
+.. code-block:: lua
+
+  local event_queue = {
+    __internal_ts_last_flush    = nil,
+    http_server_address         = "",
+    http_server_port            = 8086,
+    http_server_protocol        = "http",
+    events                      = {},
+    influx_database             = "mydb",
+    influx_user                 = "",
+    influx_password             = "",
+    max_buffer_size             = 5000,
+    max_buffer_age              = 5
+  }
+  
+  function event_queue:new(o, conf)
+    o = o or {}
+    setmetatable(o, self)
+    self.__index = self
+    for i,v in pairs(conf) do
+      if self[i] and i ~= "events" and string.sub(i, 1, 11) ~= "__internal_" then
+        broker_log:info(1, "event_queue:new: getting parameter " .. i .. " => " .. v)
+        self[i] = v
+      else
+        broker_log:warning(1, "event_queue:new: ignoring parameter " .. i .. " => " .. v)
+      end
+    end
+    self.__internal_ts_last_flush = os.time()
+    broker_log:info(2, "event_queue:new: setting the internal timestamp to " .. self.__internal_ts_last_flush)
+    return o
+  end
+
+A function to add event in queue
+********************************
+
+The add function allows to aggregate data before to add these into the queue. This 
+function replaces host ID by hostname and service ID by the description. This 
+function also operates the maximum size of the queue or the timeout:
+
+.. code-block:: lua
+
+  function event_queue:add(e)
+    local metric = e.name
+    -- time is a reserved word in influxDB so I rename it 
+    if metric == "time" then
+      metric = "_"..metric
+    end
+  
+    -- retrieve objects names instead of IDs
+    local host_name = broker_cache:get_hostname(e.host_id)
+    local service_description = broker_cache:get_service_description(e.host_id, e.service_id)
```

Comment:
Here we use the cache, don't forget to specify to restart engine to fill the cache !